granary/test/testdata/comment.mf2.json

```diff
@@ -1,5 +1,5 @@
 {
-  "type": ["h-entry", "p-comment"],
+  "type": ["h-entry", "h-as-comment"],
```

Comment:
The type map previously had a mix of mf2 types (`h-*`) and the property of the parent that this thing belongs to. It sort of makes sense because the html representation wants those two things together like `class="p-comment h-entry"`. But it led to a technically incorrect mf2 json conversion (property names in the "type" field), but more importantly made it hard to convert the target of a repost. Like, objectType: "post" doesn't imply "u-repost-of h-cite".

It's a little hard to explain, maybe touch base on IRC next time we're both around?

We could probably drop `h-as-comment`, but I need to go through and find places where it checks for `h-as-*` types for conversion back to AS.